progenitor-client/Cargo.toml

```diff
@@ -9,8 +9,16 @@ description = "An OpenAPI client generator - client support"
 [dependencies]
 bytes = "1.2.1"
 futures-core = "0.3.23"
+http = "0.2"
 percent-encoding = "2.1"
 reqwest = { version = "0.11", default-features = false, features = ["json", "stream"] }
 serde = "1.0"
 serde_json = "1.0"
 serde_urlencoded = "0.7.1"
+
+# deps for websocket support
+base64 = "0.13"
+rand = "0.8"
+# reqwest wraps hyper, but does not expose a read/write socket for us to work with,
+# so we must make direct socket connections.
```

Comment:
My reqwest PR landed and added support for doing HTTP upgrades. It would mean a slightly different interface in that we could just do the upgrade in the generated code and return some type that impls `AsyncRead/AsyncWrite`. The user of the client lib could then use `tokio_tungstenite::WebSocketStream::from_raw_socket` or similar.

Reply:
That would also avoid requiring clients picking up new deps (although tbf, those are already in the dep chain even if not explicit).

Reply:
ohh, even better!
i guess pending a proper release i'll just point it at the git repo